src/agent.ts

```diff
@@ -1,92 +1,51 @@
 import {
-  BlockEvent,
   Finding,
-  Initialize,
-  HandleBlock,
-  HealthCheck,
   HandleTransaction,
-  HandleAlert,
-  AlertEvent,
   TransactionEvent,
-  FindingSeverity,
-  FindingType,
+  FindingSeverity, 
+  FindingType,    
 } from "forta-agent";
+import { FORTA_BOT_REGISTRY, BOT_DEPLOYER_ADDRESS, NEW_AGENT_FUNCTION_SIGNATURE } from "./constants";
+
+
+export default function provideHandleTransaction(
+  proxyAddress: string,
+  deployerAddress: string,
+  functionABI: string
```

Comment:
Similar to the comment about `botTxs`, the arguments could be renamed to be a bit more specific. If this were a more general bot, this _could_ be fine, but for this one, being pretty specific/niche, the names could be more specific.